osf/models/node.py

```diff
@@ -2431,3 +2413,11 @@ def set_parent_and_root(sender, instance, created, *args, **kwargs):
     if not instance.root:
         instance.root = instance.get_root()
         instance.save()
+
```

Comment:
Either remove or restore.  Moved both of these to be called after first_save because create_permission_groups needed to be called before add_creator_as_contributor, and these methods have different senders, which would need to be accounted for if they were consolidated into one.